src/Microsoft.Health.Dicom.Core/Features/Store/StoreDatasetValidator.cs

```diff
@@ -187,7 +188,12 @@ string EnsureRequiredTagIsPresent(DicomTag dicomTag)
             {
                 if (_enableDropInvalidDicomJsonMetadata)
                 {
-                    validationResultBuilder.Add(ex, item.Tag);
+                    var message = validationResultBuilder.Add(ex, item.Tag);
```

Comment:
Instead of doing for each one, if you could do it in Storeservice and pass `storeValidatorResult.InvalidTagErrors`, you don't need the other file change. LogExporter iterates through the `storeValidatorResult.InvalidTagErrors` and logs it

Reply:
InvalidTagErrors is not for indexable attributes, it includes all tag errors

Reply:
The way it can be done cleared is have one class that represents the data and have conversion methods on the class to either tracing or other needs